src/stores/store.ts

```diff
@@ -2,13 +2,21 @@ import { create } from 'zustand';
 import { persist, createJSONStorage } from 'zustand/middleware';
 
 type Store = {
+  dataTable: object;
+  setDataTable: (data: object) => void;
+  columns: object;
+  initializeColumns: (data: object) => void;
   currentView: string;
   setCurrentView: (view: string) => void;
 };
 
 const useStore = create<Store>()(
   persist(
     (set) => ({
+      dataTable: {},
+      setDataTable: (data: object) => set({ dataTable: data }),
+      columns: {},
+      initializeColumns: (data: object) => set({ columns: data }),
       currentView: 'landing',
       setCurrentView: (view: string) => set({ currentView: view }),
```

Comment:
I'd say let's get started with organizing this store from the start before it gets too unwieldy. There are two sections here so far: 1) Table stuff and 2) Navigation stuff. I'd say add a comment above each for now and make an empty line between them to structure this more clearly.